### PR TITLE
token downscoping for folders endpoints

### PR DESCRIPTION
**Pull request by dtrodger**
- Adds folder API endpoint matching to determine a downscope token request resource attribute value

### CODE REVIEW

File: src/main/java/com/box/sdk/BoxAPIConnection.java [hunk 1]
```diff
@@ -843,10 +843,11 @@ protected ResourceLinkType determineResourceLinkType(String resourceLink) {
         try {
             URL validUrl = new URL(resourceLink);
             String validURLStr = validUrl.toString();
-            final String apiEndpointPattern = "https://api.box.com/2.0/files/\\d+";
+            final String apiFilesEndpointPattern = "https://api.box.com/2.0/files/\\d+";
```
Comment (PJSimon) [inline src/main/java/com/box/sdk/BoxAPIConnection.java:846]:
```suggestion
            final String apiFilesEndpointPattern = ".*box.com/2.0/files/\\d+";
```
I didn't catch this before.  I think we should we allow for more flexibility with the subdomains, right?

File: src/main/java/com/box/sdk/BoxAPIConnection.java [hunk 2]
```diff
@@ -843,10 +843,11 @@ protected ResourceLinkType determineResourceLinkType(String resourceLink) {
...
+            final String apiFoldersEndpointPattern = "https://api.box.com/2.0/folders/\\d+";
```
Comment (PJSimon) [inline src/main/java/com/box/sdk/BoxAPIConnection.java:847]:
```suggestion
            final String apiFoldersEndpointPattern = ".*box.com/2.0/folders/\\d+";
```
I didn't catch this before.  I think we should we allow for more flexibility with the subdomains, right?

**Comment by PJSimon** [comment]
@dtrodger Build failing due to CheckStyle complaint...

**Comment by grey-lovelace** [comment]
@PJSimon is there anything else needed on this PR before merge? This issue is blocking my team from updating to the latest release, which we need for passing in the fields param to metadataQuery.

**Comment by PJSimon** [comment]
@grey-lovelace Sorry for the delay!  I'll publish a release shortly!  Thanks!

**Comment by grey-lovelace** [comment]
@PJSimon thank you! Appreciate it.

**Pull request merged**